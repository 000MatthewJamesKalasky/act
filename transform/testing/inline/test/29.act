function bar (int<1> arg) : int<1>
{

  int<1> f;

  chp {
   self := arg + 2
  }
}

defproc test ()
{
  
  chan(int<1>) ff, qq;

  dataflow {
    bar(bar(qq)) -> ff
  }
}

test t;
