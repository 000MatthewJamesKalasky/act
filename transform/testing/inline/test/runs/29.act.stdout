defproc test ();
function bar (int<1> arg) : int<1>;

defproc test ()
{

/* instances */
chan(int<1>) ff;
chan(int<1>) qq;

/* connections */
dataflow {
qq+0x2+0x2 -> ff
}
}

function bar (int<1> arg) : int<1>
{

/* instances */
int<1> f;

/* connections */
chp {
self:=arg+0x2
}
}


/* instances */
test t;

/* connections */
